apps/mobile/src/features/account-selector-sheet.tsx

```diff
@@ -0,0 +1,88 @@
+import { RefObject, useCallback, useMemo, useRef } from 'react';
+import { ScrollView } from 'react-native';
+import { useSharedValue } from 'react-native-reanimated';
+
+import { getAvatarIcon } from '@/components/avatar-icon';
+import { Modal } from '@/components/bottom-sheet-modal';
+import { Draggable } from '@/components/draggable';
+import { AccountCard } from '@/components/wallet-settings/account-card';
+import { APP_ROUTES } from '@/constants';
+import { useAccounts, userUpdatesAccountOrder } from '@/state/accounts/accounts.slice';
+import { destructAccountIdentifer, useAppDispatch } from '@/state/utils';
+import { BottomSheetModal, BottomSheetScrollView } from '@gorhom/bottom-sheet';
+import { useRouter } from 'expo-router';
+
+import { Box } from '@leather.io/ui/native';
+
+export function AccountSelectorSheet({ modalRef }: { modalRef: RefObject<BottomSheetModal> }) {
+  const scrollViewRef = useRef<ScrollView>(null);
+  const placeholderIdx = useSharedValue<null | number>(null);
+  const accounts = useAccounts().list;
+  const direction = useSharedValue<'down' | 'up'>('down');
+  const router = useRouter();
+  const dispatch = useAppDispatch();
+  const checkIdxWithinBounds = useCallback(
+    (id: number) => {
+      return id >= 0 && id < accounts.length;
+    },
+    [accounts.length]
+  );
+
+  const accountIds = useMemo(() => accounts.map(acc => acc.id), [accounts]);
+  const swapAccountIndexes = useCallback(
+    (idx1: number | null, idx2: number | null) => {
+      if (
+        idx1 === null ||
+        idx2 === null ||
+        !checkIdxWithinBounds(idx1) ||
+        !checkIdxWithinBounds(idx2)
+      )
+        return;
+      const temp = accountIds[idx1];
+      /** @ts-expect-error we know that both indexes have both value in accounts array */
+      accountIds[idx1] = accountIds[idx2];
+      /** @ts-expect-error we know that both indexes have both value in accounts array */
```

Comment:
Any way we can avoid doing this with some runtime checks? Would say ts-expect-error isn't something we want to rely on too much